wolfcrypt/src/pkcs7.c

```diff
@@ -1298,21 +1298,170 @@ static int EncodeAttributes(EncodedAttrib* ea, int eaSz,
 }
 
 
-static int FlattenAttributes(byte* output, EncodedAttrib* ea, int eaSz)
+typedef struct FlatAttrib {
+    byte* data;
+    word32 dataSz;
+} FlatAttrib;
+
+
+/* Free FlatAttrib array and memory allocated to internal struct members */
+static void FreeAttribArray(PKCS7* pkcs7, FlatAttrib** arr, int rows)
 {
-    int i, idx;
+    int i;
+
+    if (arr) {
+        for (i = 0; i < rows; i++) {
+            if (arr[i]) {
+                if (arr[i]->data) {
+                    XMEMSET(arr[i]->data, 0, arr[i]->dataSz);
```

Comment:
If this is to zeroize why isn't ForceZero() used?

Reply:
Switched, thanks!